agnpy/radiative_process/tests/test_time_evolution.py

```diff
@@ -0,0 +1,301 @@
+
+        assert u.isclose(
+            gamma_before_synch(evaluated_n_e.gamma_min, time),
+            gamma_min,
+            rtol=0.05
+        )
+        assert u.isclose(
+            gamma_before_synch(evaluated_n_e.gamma_max, time),
+            gamma_max,
+            rtol=0.05
+        )
+        assert u.isclose(
+            evaluated_n_e.integrate(evaluated_n_e.gamma_min, evaluated_n_e.gamma_max),
+            power_law_integral(k, p, gamma_min, gamma_max),
+            rtol=0.05
+        )
+        assert u.isclose(
+            # synchrotron losses are highest at the highest energy, so test the highest energy range, as the most affected
+            evaluated_n_e.integrate(evaluated_n_e.gamma_max/10, evaluated_n_e.gamma_max),
+            power_law_integral(k, p, gamma_before_synch(evaluated_n_e.gamma_max/10, time), gamma_max),
+            rtol=0.05
+        )
+
+    def test_compare_calculation_with_calculation_split_into_two(self):
+        initial_n_e = BrokenPowerLaw(
+            k=1e-8 * u.Unit("cm-3"),
+            p1=1.9,
+            p2=2.6,
+            gamma_b=1e4,
+            gamma_min=10,
+            gamma_max=1e6,
+            mass=m_e,
+        )
+        blob = Blob(n_e=initial_n_e)
+        synch = Synchrotron(blob)
+
+        # iterate over 60 s in 20 steps
+        TimeEvolution(blob, 60 * u.s, synchrotron_loss(synch)).eval_with_fixed_intervals(intervals_count=20)
+        eval_1 = deepcopy(blob.n_e)
+        # iterate first over 30 s, and then, starting with interpolated distribution, over the remaining 30 s,
+        # with slightly different number of subintervals
+        blob.n_e = initial_n_e
+        TimeEvolution(blob, 30 * u.s, synchrotron_loss(synch)).eval_with_fixed_intervals(intervals_count=10)
+        TimeEvolution(blob, 30 * u.s, synchrotron_loss(synch)).eval_with_fixed_intervals(intervals_count=8)
+        eval_2 = blob.n_e
+
+        gamma_min = eval_1.gamma_min
+        gamma_max = eval_1.gamma_max
+        gammas = np.logspace(np.log10(gamma_min), np.log10(gamma_max))
+        assert u.allclose(
+            eval_1.evaluate(gammas, 1, gamma_min, gamma_max),
+            eval_2.evaluate(gammas, 1, gamma_min, gamma_max),
+            0.001)
+
+    def test_total_number_of_electrons_should_stay_after_time_evolution(self):
+        r_b = 1e16 * u.cm
+        n_e = PowerLaw.from_total_energy(
+            1e48 * u.erg,
+            4 / 3 * np.pi * r_b ** 3,
+            p=2.0,
+            gamma_min=1e2,
+            gamma_max=1e7,
+            mass=m_e,
+        )
+
+        blob = Blob(r_b, n_e=n_e)
+        synch = Synchrotron(blob)
+        ssc = SynchrotronSelfCompton(blob)
+
+        time = 6 * u.s
+        intervals = 300
+
+        electrons_before = n_e.integrate()
+
+        TimeEvolution(blob, time, synchrotron_loss(synch)).eval_with_fixed_intervals(
+                                                      intervals_count=intervals, max_change_per_interval=0.5)
```

Comment:
the change by 50% is already pretty extreme. This is just a test, but still it would be better to have it in a more realistic setup, maybe 10% or so (also for the lines below)